File: source/kCura.Relativity.Export/Process/ExportStatistics.vb
Namespace kCura.Relativity.Export.Process

	Public Class ExportStatistics
		Inherits Statistics
		Public Overrides Function ToDictionary() As IDictionary
			Dim retval As New System.Collections.Specialized.HybridDictionary
			If Not Me.FileTime = 0 Then retval.Add("Average file transfer rate", ToFileSizeSpecification(Me.FileBytes / (Me.FileTime / 10000000)) & "/sec")
			If Not Me.MetadataTime = 0 AndAlso Not Me.MetadataBytes = 0 Then retval.Add("Average metadata transfer rate (includes SQL processing)", ToFileSizeSpecification(Me.MetadataBytes / (Me.MetadataTime / 10000000)) & "/sec")
			Return retval
		End Function
	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Process/ExportStatistics.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
namespace kCura.Relativity.Export.Process
{

	public class ExportStatistics : Statistics
	{
		public override IDictionary ToDictionary()
		{
			System.Collections.Specialized.HybridDictionary retval = new System.Collections.Specialized.HybridDictionary();
			if (!(this.FileTime == 0))
				retval.Add("Average file transfer rate", ToFileSizeSpecification(this.FileBytes / ((double)this.FileTime / 10000000)) + "/sec");
			if (!(this.MetadataTime == 0) && !(this.MetadataBytes == 0))
				retval.Add("Average metadata transfer rate (includes SQL processing)", ToFileSizeSpecification(this.MetadataBytes / ((double)this.MetadataTime / 10000000)) + "/sec");
			return retval;
		}
	}
}
